builders/xcm/fees.md

```diff
@@ -0,0 +1,230 @@
+
+To exchange the assets for compute time, the `BuyExecution` instruction is used. Then with any remaining assets the `DepositAsset` instruction can be used to send the assets to a beneficiary.
+
+The following is an overview of each of these instructions:
+
+- **`WithdrawAsset(assets: MultiAssets)`** - removes the on-chain asset(s) and holds them in the Holding Register, a temporary position in the Cross-Consensus Virtual Machine (XCVM), which can then be used to pay for XCM execution fees
+- **`ReserveAssetDeposited(assets: MultiAssets)`** - accrues the asset(s) which have been received into the sovereign account of the local consensus system into holding
+- **`ClearOrigin`** - clears the origin to ensure that later instructions cannot command the authority of the original origin. Useful in cases where the instructions are being relayed from an untrusted source, as often is the case with `ReserveAssetDeposited`
+- **`BuyExecution(fees: MultiAsset, weight_limit: Option<Weight>)`** - takes asset(s) from holding to pay for execution fees. The amount that is actually used is determined by the interpreting system. If the fees are more than the specified amount then the instruction will result in error. You can optionally provide the amount of weight to be purchased. If the given amount of weight is lower than the estimated weight, an error will be thrown
+- **`DepositAsset(assets: MultiAssetFilter, max_assets: Compact, beneficiary: MultiLocation`** - removes asset(s) from holding to be sent to a destination account. The maximum assets defines the maximum number of unique assets to remove from holding, any remaining assets stay in holding
+
+## Calculation of Fees for Native Assets {: #calculation-of-fees-native }
+
+Substrate has introduced a weight system that determines how heavy an extrinsic is. When it comes to paying fees, users will pay a transaction fee that is proportionate to the weight of the call that is being made. One unit of weight is defined as one picosecond of execution time.
+
+The following sections will break down how to calculate XCM fees for Polkadot, Kusama, and Moonbeam-based networks. It's important to note that Kusama uses benchmarking and database read and write weight costs to determine the total weight costs for XCM instructions. Whereas Polkadot and Moonbeam-based networks use a fixed weight cost in their calculations. 
+
+There are two databases available in Polkadot and Kusama, RocksDB (which is the default) and ParityDB, both of which have their own associated weight costs for each network. 
+
+### Polkadot {: #polkadot }
+
+As previously mentioned, Polkadot currently uses a fixed amount of weight for all XCM instructions, which is `1,000,000,000` weight units.
```

Comment:
Add link to where the value comes from: https://github.com/paritytech/polkadot/blob/0f4c235e0927ab1f52d517133030af112b098bda/runtime/polkadot/src/xcm_config.rs#L95

Reply:
Oh I think this part of the review is from before I pushed changes earlier this morning because I added in a bunch of links